.github/workflows/ql-for-ql-build.yml

```diff
@@ -0,0 +1,160 @@
+name: Run QL for QL
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  queries:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache queries
+        id: cache-queries
+        uses: actions/cache@v2
+        with:
+          path: ${{ runner.temp }}/query-pack.zip
+          key: ${{ runner.os }}-queries-${{ hashFiles('**/*.ql*') }}
+      - name: Find codeql
+        if: steps.cache-queries.outputs.cache-hit != 'true'
+        id: find-codeql
+        uses: github/codeql-action/init@esbena/ql
```

Comment:
https://github.com/github/codeql-action/pull/779 should ideally get merged, but this works.